File: KULIAH/LayoutWithJava/app/src/main/java/id/ac/unsyiah/android/layoutwithjava/MainActivity.java
package id.ac.unsyiah.android.layoutwithjava;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    ViewGroup buttonContainer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        buttonContainer = findViewById(R.id.layoutCoba);
    }

    public void clickHandler(View view) {
        Button tombol = new Button(this);
        tombol.setText("Click me!");
        buttonContainer.addView(tombol);

    }
}

File: Kotlin/ColorMyViews/app/src/main/java/com/bnawan/colormyviews/MainActivity.kt
package com.bnawan.colormyviews

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setListeners()
    }

    private fun setListeners() {
        val clickableViews: List<View> =
            listOf(
                box_one_text,
                box_two_text,
                box_three_text,
                box_four_text,
                box_five_text,
                constraint_layout,
                btn_red,
                btn_yellow,
                btn_green
            )
        for (item in clickableViews) {
            item.setOnClickListener {
                makeColored(it)
            }
        }
    }

    private fun makeColored(view: View) {
        when (view.id) {

            // Boxes using Color class colors for background
            R.id.box_one_text -> view.setBackgroundColor(Color.DKGRAY)
            R.id.box_two_text -> view.setBackgroundColor(Color.GRAY)

            // Boxes using Android color resources for background
            R.id.box_three_text -> view.setBackgroundResource(android.R.color.holo_green_light)
            R.id.box_four_text -> view.setBackgroundResource(android.R.color.holo_green_dark)
            R.id.box_five_text -> view.setBackgroundResource(android.R.color.holo_green_light)

            R.id.btn_red-> box_three_text.setBackgroundResource(R.color.red)
            R.id.btn_yellow -> box_four_text.setBackgroundResource(R.color.green)
            R.id.btn_green -> box_five_text.setBackgroundResource(R.color.blue)
            else -> view.setBackgroundColor(Color.LTGRAY)
        }
    }
}